src/components/GraphiQL/GraphiQL.tsx

```diff
@@ -0,0 +1,346 @@
+    defaultSizeRelation: 1 / 3,
+    direction: "horizontal",
+    initiallyHidden: pluginContext?.visiblePlugin ? undefined : "first",
+    sizeThresholdSecond: 200,
+    storageKey: "docExplorerFlex",
+  });
+  const editorResize = useDragResize({
+    direction: "horizontal",
+    storageKey: "editorFlex",
+  });
+  const editorToolsResize = useDragResize({
+    defaultSizeRelation: 3,
+    direction: "vertical",
+    sizeThresholdSecond: 60,
+    storageKey: "secondaryEditorFlex",
+  });
+
+  const children = React.Children.toArray(props.children);
+
+  const toolbar = children.find(child =>
+    isChildComponentType(child, GraphiQL.Toolbar),
+  ) || (
+    <>
+      <ToolbarButton
+        onClick={() => prettify()}
+        label="Prettify query (Shift-Ctrl-P)"
+      >
+        <PrettifyIcon className="graphiql-toolbar-icon" aria-hidden="true" />
+      </ToolbarButton>
+      <ToolbarButton onClick={() => copy()} label="Copy query (Shift-Ctrl-C)">
+        <CopyIcon className="graphiql-toolbar-icon" aria-hidden="true" />
+      </ToolbarButton>
+      {props.toolbar?.additionalContent || null}
+    </>
+  );
+
+  const onClickReference = () => {
+    if (pluginResize.hiddenElement === "first") {
+      pluginResize.setHiddenElement(null);
+    }
+  };
+
+  const theme = useTheme();
+  const rootStyle = {
+    "--font-size-body": theme.typography.body2.fontSize,
+    "--font-size-h2": theme.typography.h3.fontSize,
+    "--font-size-h3": theme.typography.h3.fontSize,
+    "--font-size-h4": theme.typography.h4.fontSize,
+    "--font-size-hint": theme.typography.caption.fontSize,
+    "--font-size-inline-code": theme.typography.caption.fontSize,
+  } as React.CSSProperties;
+
+  const { theme: graphiqlTheme, setTheme: setGraphiqlTheme } =
+    useGraphiqlTheme();
+  useEffect(() => {
+    if (theme.themeType !== graphiqlTheme) {
+      setGraphiqlTheme(theme.themeType);
+    }
+  });
+
+  return (
+    <div
+      data-test-id="graphiql-container"
+      className="graphiql-container"
+      style={rootStyle}
+    >
+      <div className="graphiql-sidebar">
+        <div className="graphiql-sidebar-section">
+          {pluginContext?.plugins.map(plugin => {
+            const isVisible = plugin === pluginContext.visiblePlugin;
+            const label = `${isVisible ? "Hide" : "Show"} ${plugin.title}`;
+            const Icon = plugin.icon;
+            return (
+              <Tooltip key={plugin.title} label={label}>
+                <UnStyledButton
+                  type="button"
+                  className={isVisible ? "active" : ""}
+                  onClick={() => {
+                    if (isVisible) {
+                      pluginContext.setVisiblePlugin(null);
+                      pluginResize.setHiddenElement("first");
+                    } else {
+                      pluginContext.setVisiblePlugin(plugin);
+                      pluginResize.setHiddenElement(null);
+                    }
+                  }}
```

Comment:
**suggestion**: move it to the function (easier for further unit testing), also if the handler is dependent only on the pluginContext - this also can be delegated to the dedicated hook